src/command_modules/azure-cli-cdn/azure/cli/command_modules/cdn/_help.py

```diff
@@ -0,0 +1,160 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=unused-import
+from azure.cli.core.help_files import helps
+
+# pylint: disable=line-too-long
+helps['cdn'] = """
+    type: group
+    short-summary: Manage Azure Content Delivery Networks (CDN)
+"""
+
+helps['cdn profile create'] = """
+    type: command
+    examples:
+        - name: Create a CDN profile using Verizon premium CDN
+          text: >
+            az cdn profile create -g group -n profile --sku Premium_Verizon
+"""
+
+helps['cdn profile update'] = """
+    type: command
+    short-summary: Updates a CDN profile with a profile name under the specified resource group.
+"""
+
+helps['cdn profile delete'] = """
+    type: command
+    examples:
+        - name: Delete a CDN profile
+          text: >
+            az cdn profile create -g group -n profile
+"""
+
+helps['cdn profile list'] = """
+    type: command
+    short-summary: List your Azure Content Delivery Network (CDN) profiles
+    examples:
+        - name: List CDN profiles
+          text: >
+            az cdn profile list
+"""
+
+helps['cdn endpoint create'] = """
+    type: command
+    examples:
+        - name: Create an endpoint with a custom domain origin
+          text: >
+            az cdn endpoint create -g group -n endpoint --profile-name profile \\
+                --origin www.example.com
+        - name: Create an endpoint with a custom domain origin with HTTP and HTTPS ports
+          text: >
+            az cdn endpoint create -g group -n endpoint --profile-name profile \\
+                --origin www.example.com 88 4444
+        - name: Create an endpoint with a custom domain with compression and only HTTPS
+          text: >
+            az cdn endpoint create -g group -n endpoint --profile-name profile \\
+                --origin www.example.com --no-http --enable-compression
```

Comment:
I would like `--origin` to act the same way no matter what you put in for `--no-http`. I think it should stay a three tuple with a clear definition of parameters and order. I would be a bit surprised if positional parameters switched order / requirement.